cloud/services/inboundnatrules/inboundnatrules.go

```diff
@@ -101,3 +88,31 @@ func (s *Service) Delete(ctx context.Context) error {
 	}
 	return nil
 }
+
+func (s *Service) natRuleExists(ports map[int32]struct{}) func([]network.InboundNatRule, string) bool {
+	return func(rules []network.InboundNatRule, name string) bool {
+		for _, v := range rules {
+			if to.String(v.Name) == name {
+				s.Scope.V(2).Info("NAT rule already exists", "NAT rule", name)
+				return true
+			}
+			ports[*v.InboundNatRulePropertiesFormat.FrontendPort] = struct{}{}
```

Comment:
this is not correct, ordering matters now that we moved the return

Reply:
should be before the if condition

Reply:
It doesn't because if the rule exists we don't care about existing ports since we return early from the function so that set would be incomplete anyways

Reply:
I moved it down here to make it explicit, I can add a code comment too.

// if this isn't the rule we're looking for, add the port to the set of used port

Reply:
ah, my mistake, I see what you mean